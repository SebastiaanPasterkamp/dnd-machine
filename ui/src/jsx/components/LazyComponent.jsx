import React from 'react';

class LazyComponent extends React.Component
{
    shouldComponentUpdate(nextProps, nextState) {
        return (
            this.props != nextProps
            || this.state != nextState
        );
    }

    render() {
        return null;
    }
};

export default LazyComponent;