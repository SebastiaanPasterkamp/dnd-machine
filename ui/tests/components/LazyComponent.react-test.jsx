import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import LazyComponent from '../../src/jsx/components/LazyComponent.jsx';

class TestComponent extends LazyComponent
{
    constructor(props) {
        super(props);
        this.state = {
            foo: 'bar'
        };
    }
}

describe('Component: LazyComponent', () => {
    it('should not render anything', () => {
        const tree = renderer.create(
            <LazyComponent />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should not rerender because of props', () => {
        const props = {
            foo: 'bar'
        };
        const lc = shallow(
            <LazyComponent {...props}/>
        ).instance();

        const shouldUpdate = lc.shouldComponentUpdate(props, {});
        expect(shouldUpdate).toBe(false);
    });

    it('should rerender because of props', () => {
        let props = {
            foo: 'bar'
        };
        const lc = shallow(
            <LazyComponent {...props}/>
        ).instance();

        props.foo = 'foobar';
        const shouldUpdate = lc.shouldComponentUpdate(props, {});
        expect(shouldUpdate).toBe(true);
    });

    it('should not rerender because of state', () => {
        const props = {
            foo: 'bar'
        };
        const lc = shallow(
            <TestComponent {...props}/>
        ).instance();

        const shouldUpdate = lc.shouldComponentUpdate(
            props, {foo: 'bar'}
        );
        expect(shouldUpdate).toBe(false);
    });

    it('should rerender because of state', () => {
        let props = {
            foo: 'bar'
        };
        const lc = shallow(
            <TestComponent {...props}/>
        ).instance();

        props.foo = 'foobar';
        const shouldUpdate = lc.shouldComponentUpdate(
            props, {foo: 'barfoo'}
        );
        expect(shouldUpdate).toBe(true);
    });
});
