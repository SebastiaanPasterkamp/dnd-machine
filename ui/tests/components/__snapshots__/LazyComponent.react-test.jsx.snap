// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LazyComponent should not render anything 1`] = `null`;
